Jasmine/Common/ViewControllers/PhrasesExplorerViewController.swift

```diff
@@ -36,28 +41,33 @@ class PhrasesExplorerViewController: UIViewController {
         phrasesTable.didMove(toParentViewController: self)
     }
 
-    /// Dismisses this current screen and executes the callback when "Back" button is pressed.
-    @IBAction func onBackPressed(_ sender: UIBarButtonItem) {
-        onDismiss?(viewModel.selectedPhrases)
-        self.dismiss(animated: true, completion: nil)
+    /// Dismisses this current screen without saving when "Back" button is pressed.
+    /// Asks for conformation if selected phrases has been modified
+    @IBAction private func onBackButtonPressed(_ sender: UIBarButtonItem) {
+        if viewModel.hasChangedSelectedPhrases {
+            showExitWithoutSavingAlert()
+        } else {
+            self.dismiss(animated: true)
+        }
+
     }
 
-    /// Displays the level selection screen upon import pressed.
-    @IBAction func onImportPressed(_ sender: UIBarButtonItem) {
-        // TODO: Complete this when level import is done.
+    /// Dismisses this current screen and executes the callback when "Save" button is pressed.
+    @IBAction private func onSaveButtonPressed(_ sender: UIBarButtonItem) {
+        onSaveCallBack?(viewModel.selectedPhrases)
+        self.dismiss(animated: true)
     }
 
     /// Injects the required data before opening this view.
     ///
-    /// - Parameter viewModel: the view model of this class.
-    /// - Parameter isMarkable: tells the VC whether the view can be marked (given checkmark) or not.
-    /// - Parameter onDismiss: callback to execute when this view is dismissed, 
-    ///                        passes the selected phrases to the callback function
-    func segueWith(_ viewModel: PhrasesExplorerViewModel, isMarkable: Bool,
-                   onDismiss: ((Set<Phrase>) -> Void)? = nil) {
+    /// - Parameters:
+    ///   - viewModel: the view model of this class.
+    ///   - onSaveCallBack: Callback to execute when the save button is pressed. 
+    ///                     Passes the selected phrases to the callback function.
+    ///                     Phrases are selectable only if this is set.
+    func segueWith(_ viewModel: PhrasesExplorerViewModel, onSaveCallBack: ((Set<Phrase>) -> Void)? = nil) {
```

Comment:
could have `segueForViewingWith` and `segueForChoosingWith`, and the latter will require a non-nil callback.